domain/miningmanager/miningmanager_test.go

```diff
@@ -0,0 +1,446 @@
+		if err != nil {
+			t.Fatalf("Error in createParentAndChildrenTransaction: %v", err)
+		}
+		err = miningManager.ValidateAndInsertTransaction(transactionNotAnOrphan, true)
+		if err != nil {
+			t.Fatalf("ValidateAndInsertTransaction: %v", err)
+		}
+		transactionsFromMempool = miningManager.AllTransactions()
+		if !contains(transactionNotAnOrphan, transactionsFromMempool) {
+			t.Fatalf("Missing transaction %v in the mempool", transactionNotAnOrphan)
+		}
+	})
+}
+
+//	TestInsertDoubleTransactionsToMempool verifies that an attempt to insert a transaction
+//	more than once into the mempool will result in raising an appropriate error.
+func TestInsertDoubleTransactionsToMempool(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+
+		factory := consensus.NewFactory()
+		tc, teardown, err := factory.NewTestConsensus(params, false, "TestInsertDoubleTransactionsToMempool")
+		if err != nil {
+			t.Fatalf("Error setting up TestConsensus: %+v", err)
+		}
+		defer teardown(false)
+
+		miningFactory := miningmanager.NewFactory()
+		miningManager := miningFactory.NewMiningManager(tc, blockMaxMass, false)
+		transaction := createTransactionWithUTXOEntry(t, params, 0)
+		err = miningManager.ValidateAndInsertTransaction(transaction, true)
+		if err != nil {
+			t.Fatalf("ValidateAndInsertTransaction: %v", err)
+		}
+		err = miningManager.ValidateAndInsertTransaction(transaction, true)
+		if err == nil || !strings.Contains(err.Error(), "already have transaction") {
+			t.Fatalf("ValidateAndInsertTransaction: %v", err)
+		}
+	})
+}
+
+// TestHandleNewBlockTransactions verifies that all the transactions in the block were successfully removed from the memory pool.
+func TestHandleNewBlockTransactions(t *testing.T) {
+	testutils.ForAllNets(t, true, func(t *testing.T, params *dagconfig.Params) {
+
+		factory := consensus.NewFactory()
+		tc, teardown, err := factory.NewTestConsensus(params, false, "TestHandleNewBlockTransactions")
+		if err != nil {
+			t.Fatalf("Error setting up TestConsensus: %+v", err)
+		}
+		defer teardown(false)
+
+		miningFactory := miningmanager.NewFactory()
+		miningManager := miningFactory.NewMiningManager(tc, blockMaxMass, false)
+		transactionsToInsert := make([]*externalapi.DomainTransaction, 10)
+		for i := range transactionsToInsert[(coinbaseIndex + 1):] {
+			transaction := createTransactionWithUTXOEntry(t, params, i)
+			transactionsToInsert[i+1] = transaction
+			err = miningManager.ValidateAndInsertTransaction(transaction, true)
+			if err != nil {
+				t.Fatalf("ValidateAndInsertTransaction: %v", err)
+			}
+		}
+
+		_, err = miningManager.HandleNewBlockTransactions(transactionsToInsert[0:3])
+		if err != nil {
+			t.Fatalf("HandleNewBlockTransactions: %v", err)
+		}
+		mempoolTransactions := miningManager.AllTransactions()
+		for _, RemovedTransaction := range transactionsToInsert[(coinbaseIndex + 1):3] {
```

Comment:
Please change to removedTransaction, because this is not public

Reply:
My mistake, it had a prefix that I deleted and forgot to change.